### PR TITLE
Add support for u-blox EVK-ANNA-B4 which uses the nRF52833.

### PR DESCRIPTION
**Pull request by perssonmagnus**
This board is similar to the nrf52833dk_nrf52833 with
different pin assignments.

Tested with blinky, button, and Bluetooth peripheral_hr.

Signed-off-by: Magnus Persson <magnus.persson@u-blox.com>

### CODE REVIEW

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+   .. code-block:: console
+
+      $ cat ubx_evkannab4_nrf52833.overlay
+      &uart1 {
+        compatible = "nordic,nrf-uarte";
+        current-speed = <115200>;
+        status = "okay";
+        tx-pin = <14>;
+        rx-pin = <16>;
+      };
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:179]:
```suggestion
   .. code-block:: devicetree

      &uart1 {
        compatible = "nordic,nrf-uarte";
        current-speed = <115200>;
        status = "okay";
        tx-pin = <14>;
        rx-pin = <16>;
      };
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+   In the overlay file above, pin P0.16 is used for RX and P0.14 is used for TX
+
+2. Use the UART1 as ``device_get_binding(DT_LABEL(DT_NODELABEL(uart1)))``
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:183]:
```suggestion
2. Get a reference to the UART1 instance using ``DEVICE_DT_GET(DT_NODELABEL(uart1))``
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+See :ref:`set-devicetree-overlays` for further details.
+
+Selecting the pins
+==================
+To select the pin numbers for tx-pin and rx-pin:
+
+.. code-block:: console
+
+   tx-pin = <pin_no>
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:194]:
```suggestion
==================

To select the pin numbers for tx-pin and rx-pin::

   tx-pin = <pin_no>
```

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+Open the data sheet for the ANNA-B4 at `ANNA-B40 Data Sheet`_, Section 3 'Pin definition'.
+In the table 7 select the pins marked 'GPIO_xx'.  Note that pins marked as 'Radio sensitive pin'
+can only be used in under-10KHz applications. They are not suitable for 115200 speed of UART.
+
+Translate 'Pin' into number for Device tree by using the following formula::
+
+   pin_no = b\*32 + a
+
+where ``a`` and ``b`` are from the Pin value in the table (Pb.a).
+For example, for P0.1, ``pin_no = 1`` and for P1.0, ``pin_no = 32``.
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:205]: I think this information is redundant, specially now that we'll soon move to pinctrl where port/pin notation is used

File: boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+CONFIG_BOARD_UBX_EVKANNAB4_NRF52833=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# Enable hardware stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# Enable RTT
+CONFIG_USE_SEGGER_RTT=y
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig:17]: this is a non-minimal setting, so should likely be removed (applications can enable it as needed)

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 6]
```diff
@@ -61,6 +61,7 @@
 	defined(CONFIG_BOARD_UBX_EVKNINAB1_NRF52832) || \
 	defined(CONFIG_BOARD_UBX_EVKNINAB3_NRF52840) || \
 	defined(CONFIG_BOARD_UBX_EVKNINAB4_NRF52833) || \
+	defined(CONFIG_BOARD_UBX_EVKANNAB4_NRF52833) || \
```
Comment (gmarull) [inline tests/drivers/adc/adc_api/src/test_adc.c:64]: please add this change in a separate commit

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+Applications for the ``ubx_evkannab4_nrf52833`` board configuration can be
+built and flashed in the usual way (see :ref:`build_an_application`
+and :ref:`application_run` for more details); however, the standard
+debugging targets are not currently available.
+
+Flashing
+========
+
+Build and flash applications as usual (see
+:ref:`build_an_application` and :ref:`application_run` for more details)
```
Comment (gmarull) [inline boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:120]: repeated information here

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for this addition, I have a few comments

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 1 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:179]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 2 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:183]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 3 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:194]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig (hunk 5 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/ubx_evkannab4_nrf52833_defconfig:17]: This part is similar in many of the boards (nrf52833dk_nrf52833, nrf52840dk_nrf52840 etc), should I change here or should there be a common change for all of them?

File: tests/drivers/adc/adc_api/src/test_adc.c (hunk 6 above)
Comment (perssonmagnus) [reply tests/drivers/adc/adc_api/src/test_adc.c:64]: Previously we have added a number of boards (the ones above EVKANNAB4_NRF52833) where we made this change together with adding the board folder, but I guess it is nicer to have a separate commit. Will change this.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 7 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:120]: Thanks, will change.

**Comment by perssonmagnus** [comment]
Thanks for your comments. See individual answers.

**Review by tejlmand — DISMISSED** [review]

File: tests/drivers/adc/adc_api/src/test_adc.c (hunk 6 above)
Comment (perssonmagnus) [reply tests/drivers/adc/adc_api/src/test_adc.c:64]: Done.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 7 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done.

**Comment by perssonmagnus** [comment]
Second patch set updates according to comments given for each individual comment plus removed a broken link in index.rst.

**Review by tejlmand — DISMISSED** [review]
lgtm

**Comment by mbolivar-nordic** [comment]
@perssonmagnus you will need to rebase to get green CI results; the documentation failure is not due to your PR but the fix was merged to the main branch since you last pushed.

@tejlmand please re-approve after that is done

**Comment by perssonmagnus** [comment]
@tejlmand @mbolivar-nordic Thanks! Rebased and also updated commit message slightly to confirm to contribuition guidelines.

**Review by tejlmand — DISMISSED** [review]
lgtm - reapproved

**Comment by mbolivar-nordic** [comment]
@gmarull this is green now; please review

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 1 above)
Comment (gmarull) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:174]: just fix this one, other will eventually get updated.

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 2 above)
Comment (gmarull) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst:178]: same, others will eventually get updated (we're now in the process of removing many unnecessary `device_get_binding()` usages)

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 1 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 2 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 3 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

File: boards/arm/ubx_evkannab4_nrf52833/doc/index.rst (hunk 4 above)
Comment (perssonmagnus) [reply boards/arm/ubx_evkannab4_nrf52833/doc/index.rst]: Done

**Review by tejlmand — APPROVED** [review]
lgtm - reapproved

**Comment by mbolivar-nordic** [comment]
@perssonmagnus please be aware that every time you push, your reviewers will need to re-approve.

**Comment by perssonmagnus** [comment]
> @perssonmagnus please be aware that every time you push, your reviewers will need to re-approve.

@mbolivar-nordic  yes I noticed. Sorry about that, just trying to follow https://docs.zephyrproject.org/latest/contribute/#contribution-workflow. 

Is it preferred to use the "Commit suggestion" buttons in the GUI for fixing further review comments once you have one approval? Will this remove the need for re-approval?

**Review by gmarull — CHANGES_REQUESTED** [review]
All in-tree boards have now been ported to pinctrl, so please do the same here. You can refer to https://github.com/zephyrproject-rtos/zephyr/pull/43926 for more details.

**Pull request closed without merging**